### PR TITLE
Lab 01: Introduction to Go & Flutter Ershov-Dmitriy

### PR DESCRIPTION
**Pull request by dimyix**
## Lab 01: Introduction to Go & Flutter - Ershov Dmitriy
PR with points
https://github.com/dimyix/sum25-go-flutter-course/pull/1

### What was implemented:
- [ ] Go backend component with [specific features]
- [ ] Flutter frontend component with [specific features]  
- [ ] Unit tests for all major functions
- [ ] Integration between backend and frontend
- [ ] Documentation updates

### How to test:
Backend:
cd labs\lab01\backend<calculator/taskmanager/user>
go test
Frontend:
cd labs\lab01\frontend
flutter test

### Questions/Notes:
- Any challenges faced
- Design decisions made
- Areas where feedback is needed

### Checklist:
- [ ] All tests pass (make test)
- [ ] Code passes linting (make lint)
- [ ] Both Go and Flutter components work
- [ ] README updated if necessary
- [ ] Follows course coding standards

### CODE REVIEW

**Review by BlazZ1t — APPROVED** [review]
## Peer Review for Ershov Dmitriy - Lab 1

### ✅ What's Working Well
- Overall good code, just like the task required it to be

### 🔧 Areas for Improvement
- Use more provided libraries such as regex. Don't need to reinvent the wheel

### ❓ Questions
- None

### 🧪 Testing Notes
- All tests pass

### 📝 Overall Assessment
Pretty good code overall. No hard issues, though reimplementation of email validation may fail some edge cases

File: labs/lab01/backend/user/user.go [hunk 1]
```diff
@@ -1,67 +1,95 @@
...
+	if !IsValidEmail(u.Email) {
+		return ErrInvalidEmail
+	}
+
+	return nil
+}
+
+// String returns a string representation of the user, formatted as "Name: <name>, Age: <age>, Email: <email>"
+func (u *User) String() string {
+	return "Name: " + u.Name + ", Age: " + strconv.Itoa(u.Age) + ", Email: " + u.Email
```
Comment (1r444444) [inline labs/lab01/backend/user/user.go:42]: It works fine, but using fmt.Sprintf(...) would be more readable

File: labs/lab01/frontend/lib/profile_card.dart [hunk 2]
```diff
@@ -1,42 +1,61 @@
...
+      margin: const EdgeInsets.all(16.0),
+      child: Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            // TODO: add a CircleAvatar with radius 50 and backgroundImage NetworkImage(avatarUrl!) if url is not null and text name[0].toUpperCase() if url is null
+            
+            CircleAvatar(
+              radius: 24,
```
Comment (1r444444) [inline labs/lab01/frontend/lib/profile_card.dart:29]: All ok, but in the task required radius 50, not 24

**Comment by 1r444444** [comment]
The code is clean and readable, tasks are implemented correctly. Minor issues were noted directly in code comments. Good job.
